rdrf/rdrf/forms/admin/registry_registration.py

```diff
@@ -0,0 +1,28 @@
+from django.conf import settings
+from django.forms import Form, BooleanField, ChoiceField, CharField, TextInput, Select, Textarea
+from django.utils.translation import ugettext as _
+
+_form_attrs = {
+    "class": "form-control",
+}
+BootstrapTextInput = TextInput(attrs=_form_attrs)
+BootstrapSelectInput = Select(attrs=_form_attrs)
+BootstrapTextAreaInput = Textarea(attrs=_form_attrs)
+
+
+class RegistrationAdminForm(Form):
```

Comment:
Other registries like MND will need to extend this with other notifications (like ones for primary carer registration). How will this flow work in that case ? From what I see this class and RegistryAdmin will need to be extended and RegistryAdmin.setup_registration_view will need to be overridden along with adding the new templates in settings.py. Is there a way to make this a bit simpler ?

Reply:
I've been operating under the assumption that for complex registries like MND, Angelman, etc. they will want to configure more complex notifications manually. At the moment, this is more of a once-off wizard / interface to quickly see how registration is configured and turn it on and off.

So this process doesn't necessarily need to be extended, as it's more of an abstraction on top of the existing method that helps speed up registration when we are onboarding new registries more frequently.

If there's value in extending it, then I can add a proper view class etc. (perhaps it's worth doing this anyway). You're right that those are the areas I'll need to refactor

Reply:
I think there is a value in making it extensible. We should consider the base registration flow, notifications and templates defined in TRRF as the basic out of the box solution for new registries which should also be extensible enough to minimize the changes for more complex ones such as MND or Angelman. This will reduce code duplication and serve us better in the long term where more registries will be supported.

Reply:
Ok, in that case I will start making it more extensible and modify the interface such that multiple notifications can be configured.

Should notifications / templates be customisable from this action after they've been set up, or should it point to those admin pages? And if they should be customisable, is the goal to eventually remove those other pages?

Reply:
Since we already have admin pages to customize notifications/templates I think we should keep this interface simple and only use it for setting up registration. What do you think @sztamas ?

Reply:
Yes, we should re-use the exiting pages as much as possible and just have links to them.

Reply:
We could use dynamic formsets like so:

- https://schinckel.net/2017/02/05/django-dynamic-formsets/
- https://github.com/elo80ka/django-dynamic-formset
- https://stackoverflow.com/a/8097617

However, I think it'd be better to not allow adding multiple notifications at once, unless they're the pre-configured registry-specific notifications.

We could keep all the configuration in the settings file, but as email templates can be quite lengthy, I think it'd be best to make the form & view extensible. And if necessary, we can make adjust the logic if there are more complicated aspects to configure.